Sorts/QuickSortRecursive.js

```diff
@@ -43,23 +58,55 @@ const quickSort = (inputList, low, high) => {
  * @returns {number} `pIndex` pivot index value.
  */
 const partition = (partitionList, low, high) => {
-  const pivot = partitionList[high]
+  const mid = Math.floor((low + high) / 2)
+  // get the median of three is good technique for partitioning to be sure that the 2 sub-arrays will be almost equal or nearly equal in size
```

Comment:
This comment is phrased a bit poorly and also misleading: Nothing is guaranteed here. It's just a heuristic. It can still be horribly wrong if the arrays are crafted to be worst case.

Reply:
I am with you that nothing guaranteed but trying to improve something worth it.